src/common/arrstr.cpp

```diff
@@ -721,3 +732,217 @@ wxArrayString wxSplit(const wxString& str, const wxChar sep, const wxChar escape
 
     return ret;
 }
+
+#if wxUSE_REGEX
+
+namespace // helpers needed by wxCmpNaturalGeneric()
+{
+// Used for comparison of string parts
+struct wxStringFragment
+{
+    // Fragment types are generally sorted like this:
+    // Empty < SpaceOrPunct < Digit < LetterOrSymbol
+    // Fragments of the same type are compared as follows:
+    // SpaceOrPunct - collated, Digit - as numbers using value
+    // LetterOrSymbol - lower-cased and then collated
+    enum Type
+    {
+        Empty,
+        SpaceOrPunct,  // whitespace or punctuation
+        Digit,         // a sequence of decimal digits
+        LetterOrSymbol // letters and symbols, i.e., anything not covered by the above types
+    };
+
+    wxStringFragment() : type(Empty), value(0) {}
+
+    Type     type;
+    wxString text;
+    wxUint64 value; // used only for Digit type
+};
+
+
+wxStringFragment GetFragment(wxString& text)
+{
+    static const wxRegEx reSpaceOrPunct(wxS("^([[:space:]]|[[:punct:]])+"));
+    static const wxRegEx reDigit(wxS("^[[:digit:]]+"));
+    static const wxRegEx reLetterOrSymbol("^[^[:space:]|[:punct:]|[:digit:]]+");
+
+    wxStringFragment fragment;
+    size_t           length = 0;
+
+    if ( text.empty() )
+        return fragment;
+
+    // In attempt to minimize the number of wxRegEx.Matches() calls,
+    // try to do them from the most expected to the least expected
+    // string fragment type.
+    if ( reLetterOrSymbol.Matches(text) )
+    {
+        if ( reLetterOrSymbol.GetMatch(NULL, &length) )
+        {
+            fragment.type = wxStringFragment::LetterOrSymbol;
+            fragment.text = text.Left(length);
+        }
+    }
+    else if ( reDigit.Matches(text) )
+    {
+        if ( reDigit.GetMatch(NULL, &length) )
+        {
+            fragment.type = wxStringFragment::Digit;
+
+            // The number may be too big to fit into a wxUint64,
+            // so take at most its first 19 digits, the rest of
+            // the number will be processed as the next string fragment.
+            // Not perfect, but I could not think of a better simple solution.
+            length = wxMin(length, 19);
+            fragment.text = text.Left(length);
+            fragment.text.ToULongLong(&fragment.value);
+        }
+    }
+    else if ( reSpaceOrPunct.Matches(text) )
+    {
+        if ( reSpaceOrPunct.GetMatch(NULL, &length) )
+        {
+            fragment.type = wxStringFragment::SpaceOrPunct;
+            fragment.text = text.Left(length);
+        }
+    }
+
+    text.erase(0, length);
+    return fragment;
+}
+
+int CompareFragmentNatural(const wxStringFragment& lhs, const wxStringFragment& rhs)
+{
+    int result = 1;
+
+    switch ( lhs.type )
+    {
+        case wxStringFragment::Empty:
+            switch ( rhs.type )
+            {
+                case wxStringFragment::Empty:
+                    result = 0;
+                    break;
```

Comment:
Why not just `return 0;`? There's nothing being done at the end of the outer switch, and one would need quite a bit of scrolling to find that out.

Reply:
This is basically so that the code for all cases is analogical. I believe that if we return from here, we should return from all cases.

Reply:
Yes, the same for all `case`s.
I think it will help with readability, rather than anything else.